fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/RecordQueryPlanner.java

```diff
@@ -1530,7 +1530,7 @@ private static boolean addCoveringField(@Nonnull KeyExpression requiredExpr,
     }
     
     @Nullable
-    public RecordQueryPlan planCoveringAggregateIndex(@Nonnull RecordQuery query, @Nonnull String indexName) {
+    public RecordQueryCoveringIndexPlan planCoveringAggregateIndex(@Nonnull RecordQuery query, @Nonnull String indexName) {
```

Comment:
This should be considered a binary-incompatible change of the stable API.

Reply:
I added a note in the release notes. Do we think stronger action is needed?

Reply:
@alecgrieser Do we need a release bump? It's source compatible, but would break on binary level.

Reply:
Good question. (I'm not sure if there's a settled answer on that.) I guess technically we only make guarantees about our API stability (and not our ABI stability). This...feels like it's probably fine to just make a note of, I suppose, but I'm not certain.

It seems a little weird to me that this method would return a concrete implementation of `RecordQueryPlan` rather than an interface (from an API perspective), but I've not yet gone through the code enough to understand the justification for that change yet.